app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -104,6 +105,7 @@ class EcsCapacityProvider extends React.Component<IEcsCapacityProviderProps, IEc
   private updateCapacityProviderType = (targetCapacityProviderType: boolean) => {
     this.setState({useDefaultCapacityProviders : targetCapacityProviderType});
     this.props.notifyAngular("useDefaultCapacityProviders", targetCapacityProviderType);
+    this.props.command.viewState.dirty.capacityProviders = [];
```

Comment:
What happens if there isn't a default cp defined, but I click the default cp toggle? (will it clear the message, and is that actually accurate?)

Reply:
Yes, in any case if you toggle from custom to default this message will disappear because we display default CPs coming from the API response.

Reply:
Is it possible to delete a defined default capacity provider, and could we run into the same issue there?

Reply:
When we choose custom capacity provider and if dirty capacity provider exist then we show an error and ask user to select the different CP from the dropdown. But at this point dirty CP is not a part of this dropdown options. Similarly, in default CP case we do not show already deleted CP (which are dirty in this case) hence there is no need of showing the error. User just have to click the **Done** button and pipeline will have the latest default CPs without any dirty one.

Reply:
I'm thinking more from the perspective that I'm a user, and I click edit existing server group, I scroll down, and don't see any issues, so I click cancel. As a user, how would I be aware that the default capacity provider was deleted? Also, what happens if I toggle from capacity providers to default and then back to capacity providers (does it clear the error message)?

Reply:
When you toggle from default to custom capacity provider then you need to set a new capacity provider strategy. In that case error won't be shown as you are setting a new CP and dirty CP is not a part of dropdown options.
In the case of default CP we need to figure out how do we indicate that something has been deleted and they need to click done instead of cancel.